kube-runtime/src/watcher.rs

```diff
@@ -217,6 +234,21 @@ pub struct Config {
     /// Configures re-list for performance vs. consistency.
     pub list_semantic: ListSemantic,
 
+    /// Kubernetes 1.27 Streaming Lists
+    /// Determines how the watcher fetches the initial list of objects.
+    /// Defaults to `ListWatch` for backwards compatibility.
+    ///
+    /// ListWatch: The watcher will fetch the initial list of objects using a list call.
+    /// StreamingList: The watcher will fetch the initial list of objects using a watch call.
+    ///
+    /// StreamingList is more efficient than ListWatch, but it requires the server to support
+    /// streaming list bookmarks. If the server does not support streaming list bookmarks,
+    /// the watcher will fall back to ListWatch.
+    ///
+    /// See https://kubernetes.io/docs/reference/using-api/api-concepts/#streaming-lists
+    /// See https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/3157-watch-list#design-details
```

Comment:
```suggestion

    /// Control how the watcher fetches the initial list of objects.
    ///
    /// ListWatch: The watcher will fetch the initial list of objects using a list call.
    /// StreamingList: The watcher will fetch the initial list of objects using a watch call.
    ///
    /// StreamingList is more efficient than ListWatch, but it requires the server to support
    /// streaming list bookmarks (opt-in feature gate in Kubernetes 1.27).
    ///
    /// See https://kubernetes.io/docs/reference/using-api/api-concepts/#streaming-lists
    /// See https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/3157-watch-list#design-details
```

Reworded a bit. Plus removed the line about fallback because if the server does not support it, it just errors.

```
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `/home/clux/kube/kube/target/debug/examples/namespace_reflector`
Error: watch stream failed: ApiError: ListOptions.meta.k8s.io "" is invalid: sendInitialEvents: Forbidden: sendInitialEvents is forbidden for watch unless the WatchList feature gate is enabled: Invalid (ErrorResponse { status: "Failure", message: "ListOptions.meta.k8s.io \"\" is invalid: sendInitialEvents: Forbidden: sendInitialEvents is forbidden for watch unless the WatchList feature gate is enabled", reason: "Invalid", code: 422 })

Caused by:
    0: ApiError: ListOptions.meta.k8s.io "" is invalid: sendInitialEvents: Forbidden: sendInitialEvents is forbidden for watch unless the WatchList feature gate is enabled: Invalid (ErrorResponse { status: "Failure", message: "ListOptions.meta.k8s.io \"\" is invalid: sendInitialEvents: Forbidden: sendInitialEvents is forbidden for watch unless the WatchList feature gate is enabled", reason: "Invalid", code: 422 })
    1: ListOptions.meta.k8s.io "" is invalid: sendInitialEvents: Forbidden: sendInitialEvents is forbidden for watch unless the WatchList feature gate is enabled: Invalid
```

Reply:
This comment suggestion, or something like it should still be applied. Currently this comment is on the `initial_list_strategy` member, and that does not need to be configured to use streaming lists (as that only refers to one variant).